src/plugins/size/size.c

```diff
@@ -0,0 +1,140 @@
+/**
+ * @file
+ *
+ * @brief Source for size plugin
+ *
+ * @copyright BSD License (see LICENSE.md or https://www.libelektra.org)
+ *
+ */
+
+#include "size.h"
+
+#include <kdbhelper.h>
+#include <stdio.h>
+#include <string.h>
+
+
+#define DEFAULT_SEPARATOR "x"
+
+static int transform (KeySet * appendKS, const Key * key, const char * sep)
+{
+	char * value = elektraStrDup (keyString (key));
+	if (!value) return 0;
+	char * dptr = strstr (value, sep);
+	if (!dptr) return 0;
+	*dptr = '\0';
+	Key * widthKey = keyNew (keyName (key), KEY_END);
+	keyAddBaseName (widthKey, "Width");
+	keySetString (widthKey, value);
+
+	Key * heightKey = keyNew (keyName (key), KEY_END);
+	keyAddBaseName (heightKey, "Height");
+	keySetString (heightKey, dptr + elektraStrLen (sep) - 1);
+
+	ksAppendKey (appendKS, widthKey);
+	ksAppendKey (appendKS, heightKey);
+
+	elektraFree (value);
+	return 1;
+}
+
+
+int elektraSizeClose (Plugin * handle, Key * parentKey ELEKTRA_UNUSED)
+{
+	KeySet * ks = elektraPluginGetData (handle);
+	if (ks)
+	{
+		ksDel (ks);
+	}
+	elektraPluginSetData (handle, NULL);
+	return 0;
+}
+
+int elektraSizeSet (Plugin * handle, KeySet * returned, Key * parentKey ELEKTRA_UNUSED)
+{
+	Key * cur = NULL;
+	KeySet * iterKS = elektraPluginGetData (handle);
+	if (!iterKS) return 0;
+	ksRewind (iterKS);
+	while ((cur = ksNext (iterKS)) != NULL)
+	{
+		keyDel (ksLookup (returned, cur, KDB_O_POP));
```

Comment:
I do not understand this, please add source-code comment here.

How does this code transform the width/height keys back to the original value widthxheight?
Or document the limitation if it is one.